### PR TITLE
Fix the Windows build in the CI

### PR DESCRIPTION
**Pull request by niyatim23**
*Issue #, if available:*
https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/issues/1670
https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/issues/1657
https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/issues/1624
https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/issues/1613
https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/issues/1611
https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/issues/1451
https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/issues/1288

*Description of changes:*

What was changed?

Fixed the Windows CI build which has been broken since forever. Modified the dependency's (libwebsockets) CMake file to fix the same by introducing arguments that were missing

Why was it changed?

We were missing the `OPENSSL_LIBRARIES`, instead we specified the `SSL_LIBRARY` and `CRYPTO_LIBRARY` which worked on Linux and Unix systems but was breaking on Windows. Additionally, introduced a `LIST_SEPARATOR` as the `OPENSSL_LIBRARIES` contains two file paths separated by a semicolon which causes CMake to ignore them at runtime, so I have used `|` as a separator instead

How was it changed?

By modifying the CMakeLists files

Testing:

With the CI

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by sirknightj — APPROVED** [review]
Nice work!

File: CMakeLists.txt [hunk 1]
```diff
@@ -31,6 +31,9 @@ option(THREAD_SANITIZER "Build with ThreadSanitizer." OFF)
 option(UNDEFINED_BEHAVIOR_SANITIZER "Build with UndefinedBehaviorSanitizer." OFF)
 option(LINK_PROFILER "Link gperftools profiler" OFF)
 
+set(EXT_PTHREAD_INCLUDE_DIR "" CACHE PATH "Path to the PThread dir")
```
Comment (disa6302) [inline CMakeLists.txt:34]: This would be set only for Windows right? Can we add the platform check?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -342,33 +342,51 @@ jobs:
...
+      - name: Build repository
+        shell: powershell
+        run: |
+          cd D:\a\webrtc
+          git config --system core.longpaths true
+          .github\build_windows.bat
+      # TODO: Fix these tests
+      # - name: Run tests
+      #   run: |
+      #     & "D:\a\webrtc\build\tst\webrtc_client_test.exe --gtest_filter='-DataChannelFunctionalityTest.*:IceApiTest.*:IceFunctionalityTest.*:PeerConnectionFunctionalityTest.*:SignalingApiFunctionalityTest.*:TurnConnectionFunctionalityTest.*:RtpFunctionalityTest.marshallUnmarshallH264Data:RtpFunctionalityTest.packingUnpackingVerifySameH264Frame:RtcpFunctionalityTest.onRtcpPacketCompound:RtcpFunctionalityTest.twcc3'"
```
Comment (disa6302) [inline .github/workflows/ci.yml:389]: We are not running the unit tests? Will that be focussed on another PR? If yes, can we create a github issue to track it?

**Comment by disa6302** [comment]
On another note, I think we are missing a windows CI with Mbedtls enabled. Something to track as well to include at some point. @niyatim23 , can we create a github issue for that too?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -342,33 +342,51 @@ jobs:
...
+          dir
+      - name: Install dependencies
+        shell: powershell
+        run: |
+          choco install gstreamer --version=1.16.2
+          choco install gstreamer-devel --version=1.16.2 
+          curl.exe -o C:\tools\pthreads-w32-2-9-1-release.zip ftp://sourceware.org/pub/pthreads-win32/pthreads-w32-2-9-1-release.zip
+          mkdir C:\tools\pthreads-w32-2-9-1-release\
+          Expand-Archive -Path C:\tools\pthreads-w32-2-9-1-release.zip -DestinationPath C:\tools\pthreads-w32-2-9-1-release    
+      # - name: Build libwebsockets from source
```
[the remark this answers is not in the document]
Comment (niyatim23) [reply .github/workflows/ci.yml:367]: I left those in because we may need it in the future when we upgrade to a higher version of libwebsockets

File: CMakeLists.txt (hunk 1 above)
Comment (niyatim23) [reply CMakeLists.txt:34]: Since we are allowing it to be provided from command line, I think it needs to exist like that. It is empty at first ("") and then when it is read from command line, it gets its value which is used later

Comment (disa6302) [reply CMakeLists.txt:34]: Its still a -D option right? If its other platforms, this blocks is just not visited and the variable is unused. So it should be ok to add this in the `if(WIN32)` block. It is more clear that way.

Comment (niyatim23) [reply CMakeLists.txt]: Makes sense, done!

**Comment by niyatim23** [comment]
> On another note, I think we are missing a windows CI with Mbedtls enabled. Something to track as well to include at some point. @niyatim23 , can we create a github issue for that too?

Tracking:
1. Windows-mbedTLS: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/issues/1768
2. Windows-tests: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/issues/1767

File: .github/workflows/ci.yml (hunk 2 above)
Comment (niyatim23) [reply .github/workflows/ci.yml:389]: Tracking: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/issues/1767

**Review by disa6302 — APPROVED** [review]
Great work @niyatim23 !
